subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/forsubgraph/SubgraphVersionConstraintsFeatureInteractionIntegrationTest.groovy

```diff
@@ -75,15 +75,79 @@ class SubgraphVersionConstraintsFeatureInteractionIntegrationTest extends Abstra
         resolve.expectGraph {
             root(':', ':test:') {
                 module('org:bar:1.0') {
-                    module('org:baz:1.0').byRequest()
-                    module('org:foo:1.0') {
+                    edge('org:baz:{require 1.0; subgraph}', 'org:baz:1.0').byRequest()
```

Comment:
Is that notation just for tests?
because it seems to me that a version would be printed as `{ required 1.0; subgraph true }`. Correct?

Reply:
No it is the notation used everywhere it is printed. Examples:
 `1.0` (only require)
`{require 1.0; subgraph}` (require+subgraph)
`{require 1.0; reject 2.0}` (require+reject)
`{require 1.0; reject 2.0; subgraph}` (require+reject+subgraph)

So basically we treat is as _set_ or _unset_ (print nothing). Similar to e.g. reject where we also print nothing if no reject is defined. There is a second `append` method for handling boolean values in `AbstractVersionConstraint`.

Reply:
Of course, I just failed at reading code it seems 😞 
All good then!